stripe/src/main/java/com/stripe/android/model/SourceParams.java

```diff
@@ -375,19 +388,114 @@ public String getType() {
     }
 
     /**
+     * @return a custom type of this source, if one has been set
+     */
+    @Nullable
+    public String getCustomType() {
+        return mCustomType;
+    }
+
+    /**
      * @return the custom metadata set on these params
      */
     public Map<String, String> getMetaData() {
         return mMetaData;
     }
 
+    /*---- Setters ----*/
+
+    /**
+     * @param amount currency amount for this source, in the lowest denomination.
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setAmount(long amount) {
+        mAmount = amount;
+        return this;
+    }
+
+    /**
+     * @param apiParameterMap a map of parameters specific for this type of source
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setApiParameterMap(
+            @NonNull Map<String, Object> apiParameterMap) {
+        mApiParameterMap = apiParameterMap;
+        return this;
+    }
+
+    /**
+     * @param currency currency code for this source (i.e. "EUR")
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setCurrency(String currency) {
+        mCurrency = currency;
+        return this;
+    }
+
+    /**
+     * @param owner an {@link SourceOwner} object for this source
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setOwner(Map<String, Object> owner) {
+        mOwner = owner;
+        return this;
+    }
+
+    /**
+     * Sets a redirect property map for this source object. If you only want to
+     * set a return url, use {@link #setReturnUrl(String)}.
+     *
+     * @param redirect a set of redirect parameters
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setRedirect(Map<String, Object> redirect) {
+        mRedirect = redirect;
+        return this;
+    }
+
+
+    /**
+     * @param returnUrl a redirect URL for this source.
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setReturnUrl(@NonNull @Size(min = 1) String returnUrl) {
+        setRedirect(createSimpleMap(FIELD_RETURN_URL, returnUrl));
+        return this;
+    }
+
+    /**
+     * Sets the {@link SourceType} for this source. If you are creating a custom type,
+     * use {@link #setCustomType(String)}.
+     *
+     * @param type the {@link SourceType}
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setType(@Source.SourceType String type) {
+        mType = type;
+        return this;
+    }
+
+    /**
+     * Sets a custom type for the source, and sets the {@link #mType type} for these parameters
+     * to be {@link Source#UNKNOWN}.
+     *
+     * @param customType the name of the source type
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setCustomType(@NonNull String customType) {
```

Comment:
Same comment as above – I think `setCustomType("bancontact")` should result in `type = BANCONTACT`